docs/smart-contracts/entrypoints.md

```diff
@@ -12,77 +12,55 @@ The entrypoints in a Tezos smart contract must meet these specifications:
-
-From Raphael:
-
-> When calling a contract, there is no difference between explicitly passing default as entrypoint name and passing no entrypoint at all.
-
-> When defining and originating a contract, the entrypoints are declared as annotations in the parameter type of the contract. There may or not be a %default in this parameter type. If there is no %default annotation, the whole parameter type is the type of the default entrypoint.
-
-> you can still have other entrypoints but they are special cases of what you can achieve by calling the default entrypoint.
-
-For example, if the parameter type is or (unit %be_nice) (unit %be_nasty), then calling the contract with:
-- Unit on the be_nice entrypoint is equivalent to
-- Left Unit on the default entrypoint
-and:
-- Unit on the be_nasty entrypoint is equivalent to
-- Right Unit on the default entrypoint.
-
-
-Each contract has a special entrypoint named "default," which runs when a caller calls the contract without specifying an entrypoint, such as if the caller sends tez to the contract.
-Senders do not pass parameters to this endpoint, though technically it accepts a parameter of the type `Unit`.
--->
-
+| `increment` | `increment` takes an `int` as a parameter and adds it to the previous value of the storage |
+| `decrement` | `decrement` takes an `int` as a parameter and subtracts it from the previous value of the storage |
 
-<!-- TODO old docs say "if this entrypoint exists" ; doesn't it always exist? -->
-<!--
-  - `default` doesn't take any parameter (or more specifically, its parameter is of type `Unit`). If this entrypoint exists, it will be executed any time the contract is called without specifying any entrypoint or parameter. This is in particular the case when a user or a contract simply sends some tez to the contract, as you do when sending tez to a user.
--->
+## The default entrypoint
 
+All contracts have a default entrypoint.
+This entrypoint runs when a caller calls the contract without specifying an entrypoint, such as when a caller sends tez to the contract.
+By default, the default entrypoint accepts the tez and does nothing.
 
-## Entrypoint implementation
+You can implement the default entrypoint to control what happens when a caller calls the contract without specifying an entrypoint and to allow callers to explicitly call the default entrypoint.
+For example, if you don't want the contract to accept tez, you can fail the transaction if the caller sends tez without specifying an entrypoint.
 
-Internally, entrypoints are implemented using a variant data type.
-The contract contains a single piece of logic that branches based on the value of that variant.
-In Michelson, the different values of the variant are annotated with the name of the corresponding entrypoint.
-<!-- TODO link to variants -->
+The sender does not pass parameters to the default entrypoint, although technically it accepts a parameter of the type `Unit`, which indicates no parameter.
```

Comment:
No, the default entrypoint can take a non-unit parameter.

Reply:
Updated to: "By default, the default entrypoint accepts a parameter of the type `Unit`, which indicates no parameter. If you implement the default entrypoint, it can accept any parameter that other entrypoints can accept."